src/sql/workbench/services/executionPlan/common/executionPlanService.ts

```diff
@@ -27,19 +28,37 @@ export class ExecutionPlanService implements IExecutionPlanService {
 		this._providerRegisterEvent = this._onProviderRegister.event;
 	}
 
-	private async ensureCapabilitiesRegistered(): Promise<void> {
-		let providers = Object.keys(this._capabilitiesService.providers);
-		if (!providers) {
-			await new Promise<void>(resolve => {
-				this._capabilitiesService.onCapabilitiesRegistered(e => {
-					resolve();
-				});
-			});
-		}
+	/**
+	 * This ensures that the capabilities service has registered the providers to handle execution plan requests.
+	 * @param providerId Optional provider id to wait for.
+	 */
+	private async ensureCapabilitiesRegistered(providerId?: string): Promise<void> {
+		let providers: string[] = [];
+		// Wait until the capabilities service has registered some providers.
+		await new Promise<void>(resolve => {
+			let retryCount = 0;
+			const intervalId = setInterval(() => {
+				while (retryCount < 5) {
+					providers = Object.keys(this._capabilitiesService.providers);
+					if (providers) {
+						if (providerId && providers[providerId]) {
```

Comment:
You can probably simplify this to: `if ((providerId && providers[providerId]) || providers)`

Reply:
Actually you can do that in the if that checks providers and remove a level of conditional nesting.

Reply:
that's not correct as it will resolve for condition when providerID is not found in providers but providers is not undefined.

Reply:
I have cleaned up the condition now.